scaffolding/src/datatypes/arenavec.rs

```
@@ -118,7 +129,7 @@ impl<T> ArenaVec<T> {
         }
     }
 
-    pub fn push(&self, val: T) {
+    pub fn push(&self, val: T) -> Result<()> {
```

Not sure about this... `Vec` doesn't return a result from `push` either

if we return a `Result` the compiler will emit a warning that it sound be used. So people are basically going to add an `unwrap()` to every push call. And in this case the error is really unlikely to happen... plus I'm not sure there's a better way to handle it

Just loose thoughts, I'm not really sure, but I feel like panicking instead of returning a result will be a nicer API

actually I guess we could have a `try_push` that returns a result and a `push` that just panicks?

---

That's probably the best option. I just like having fallible memory operations (I think if the rust std could be made over again, that would be the main change I'd ask for), but we can have the panicking ones to, as long as we still get `try_()`. https://lkml.org/lkml/2021/4/14/1099.